portal/content/docs/contributing/architecture.md

```diff
@@ -0,0 +1,357 @@
+9. The `types.ITableLoader` implementation uses the `table/read(path=cars.csv)` to get the tabular data source, paginate and filter it.
+
+## Design System
+
+The project uses the following principles for the design system:
+
+### Familiarity
+
+The high-level design of the Open Data Editor project is inspired by the layouts of popular text and code editors like VSCode or typical texte editor. The goal is to make the user interface familiar to users who are already familiar with these tools.
+
+### Simplicity
+
+Although the Open Data Editor project is designed after complext programming IDE, the project's UI is way simpler. The application doesn't have complex menus, toolbars, or sidebars. The main focus is on the file browser, the editor, and the metadata editor. There are less buttons in menus and they are generally bigger than in typical IDEs.
+
+### Consistency
+
+The Open Data Editor project uses a consistent dialogs, editors, views, and menus for all file types. For example, in dialogs, the rejecting action is always on the left, and the approving is on the right (see "Shortcuts" and "Color Coding" as well). One of the main user challenges of working with IDEs is to remember the controls locations. The Open Data Editor project tries to solve this problem by using a consistent layout for all file types making buttons enabled and disabled based on available functionality.
+
+## Shortcuts
+
+Using keyboard shortcuts is a key practice for making advanced users productive. The Open Data Editor project provides as many shortcuts as possible. Keyboard shortcuts are visually available on the buttons to speed user learning.
+
+## Color Coding
+
+The Open Data Editor project uses consistent color coding system accross the application. The main colors are:
+
+- **blue**: actions
+- **yellow**: visual selections
+- **orange**: attention (potentially destructive actions)
+- **green**: valid state
+- **red**: invalid state
+- **grey**: disabled
+
+## Weaknesses Analysis
+
+Here are the list of some potintial weaknesses of the Open Data Editor architecture:
+
+### Python Dependency
+
+Historically, the project relied on the `frictionless-py` framework for all tabular data operations. Although `frictionless-py` is battle-tested library it has its own limitations regarding perfomance, maintance and, generally speaking, legacy approaches that are used in its data-processing and data-modelling architecture.
+
+Using `frictionless-py` also brings the whole Python environment to be installed and used for Open Data Editor to work properly. This is a significant limitation for the project as it makes the project less portable and more complex to setup. It creates a lot of issues with the Electron environment and makes the project less stable and not possible to be run without the Internet on the initialization stage.
```

Comment:
I'm wondering if we could explore an idea to just provide a default virtual environment as an asset of the project.

Reply:
We tried to bundle `frictinless-py` and Python with tools like `py2exe` etc but it never worked. It might need more time to exploration as it would be way better if it doesn't need to install Python and deps on the user machine